File: matlab/config/config_andermatt_zuerich_pr.m
function cfg = config_andermatt_zuerich_pr()

% Variable settings
cfg.clim_var_name = 'pr';
cfg.clim_var_long_name = 'Precipitation';
cfg.clim_var_units = 'mm';
cfg.qmf_period = 'monthly'; % 'whole', 'seasonal', or 'monthly'
cfg.bc_type = 'multiplicative'; % 'additive' or 'multiplicative'
cfg.preserve_trends = true; % true or false
cfg.trend_window = 365*5; % Moving mean window in time steps
cfg.agg_method = 'sum'; % 'mean' or 'sum'
cfg.write_output = true;
cfg.n_quantiles = 1001; % Number of quantiles for QMFs
cfg.idw_power = 2; % Inverse distance weighting distance exponent
cfg.idw_method = 'elevation_aware'; % 'horizontal' or 'elevation_aware'
cfg.idw_alpha = 10; % 100 m vertical = 1000 m horizontal (1000/100 = 10)
cfg.multiplicative_epsilon = 0.1; % Offset for multiplicative detrending/retrending
cfg.use_parallel = false;
cfg.n_workers = []; % [], 2, 3 ...
cfg.keep_grid_biases = false;
cfg.coordinate_system = 'geographic'; % 'geographic' or 'projected'

% File paths
cfg.file_path_station_coords = 'N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\input_data\meteoswiss\processed\andermatt_zuerich_coordinates.csv';
cfg.file_path_station_clim_var = 'N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\input_data\meteoswiss\processed\andermatt_zuerich_pr.csv';
cfg.file_path_raw_data = 'N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\input_data\chelsa\processed\andermatt_zuerich_1981_2019\pr_andermatt_zuerich_1981_2019.nc';
cfg.file_path_bc_data = 'N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\output_data\andermatt_zuerich_1981_2019\pr_bc_andermatt_zuerich_1981_2019.nc';
cfg.file_path_figures = 'N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\output_data\figures\';

end

File: scripts/config_andermatt_zuerich_pr.py
from eqm_step import BiasCorrectionConfig


def config_andermatt_zuerich_pr() -> BiasCorrectionConfig:
    return BiasCorrectionConfig(
        clim_var_name="pr",
        clim_var_long_name="Precipitation",
        clim_var_units="mm",
        qmf_period="monthly",
        bc_type="multiplicative",
        preserve_trends=True,
        trend_window=365 * 5,
        agg_method="sum",
        write_output=True,
        n_quantiles=1001,
        idw_power=2,
        idw_method="elevation_aware",
        idw_alpha=10,
        multiplicative_epsilon=0.1,
        use_parallel=False,
        n_workers=None,
        keep_grid_biases=False,
        coordinate_system="geographic",
        file_path_station_coords=r"N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\input_data\meteoswiss\processed\andermatt_zuerich_coordinates.csv",
        file_path_station_clim_var=r"N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\input_data\meteoswiss\processed\andermatt_zuerich_pr.csv",
        file_path_raw_data=r"N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\input_data\chelsa\processed\andermatt_zuerich_1981_2019\pr_andermatt_zuerich_1981_2019.nc",
        file_path_bc_data=r"N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\output_data\andermatt_zuerich_1981_2019\pr_bc_andermatt_zuerich_1981_2019.nc",
        file_path_figures=r"N:\gebhyd\8_Him\Personal_folders\Mike\foracca\paper\output_data\figures",
    )
